drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc

```diff
@@ -617,8 +614,20 @@ function boincwork_generate_prefs_element(&$form, $type, $elements, $user_prefs
       } else {
         // If user_applications is not defined, it means there are no
         // app_ids set in the user's project preferences payload
-        // XML. Thus all Web form values should be checked.
-        $checked = TRUE;
+        // XML. Thus all Web form values should be determined by the
+        // app_overrides container. If the application is not enabled
+        // then the application is not selected. If the application is
```

Comment:
Currently the behavior is, when the user clicks Save in the Web form, the application that is disabled (but selected is true) will **not** appear in the payload XML in an  `<app_id>` element.

The behavior desired is that in this case, the application is added to the payload XML, because the user doesn't have a choice whether or not this particular application is selected.